File: Semestru 3/MAP/lab/lab2/src/container/TaskContainer.java
package container;

import model.Task;

import java.util.ArrayList;
import java.util.List;

public abstract class TaskContainer implements Container {
    List<Task> tasks = new ArrayList<>();

    @Override
    public abstract Task remove();

    @Override
    public void add(Task task) {
        this.tasks.add(task);
    }

    @Override
    public int size() {
        return this.tasks.size();
    }

    @Override
    public boolean isEmpty() {
        return this.tasks.isEmpty();
    }
}

File: Semestru 3/MAP/lab/lab9/lab9/lab9/container/TaskContainer.cs
using System;
using Task = lab9.model.Task;

namespace lab9.container
{
	internal abstract class TaskContainer : IContainerr
	{
		public List<Task> tasks = new List<Task>();
		public void Add(Task task) => tasks.Add(task);
		public bool IsEmpty() => tasks.Count == 0;
		public abstract Task? Remove();
		public int Size() => tasks.Count;
	}
}
